### ISSUE TITLE
Implement a file locking mechanism for files under registration

### ISSUE THREAD

**Issue by vartald** [bluenviron/mediamtx#3704]
I would like to propose the implementation of a file locking mechanism to indicate that a file is currently being registered and is in use within the registration directory:

- Create a lock file: At the beginning of a registration process, the system would automatically create a lock file in the same directory as the file being registered.
- Lock file naming: The lock file name could follow a specific convention (e.g., <filename>.lock) for easy identification.
- Remove the lock file: At the end of the registration process (whether successful or not), the lock file would be automatically removed.
- Checking mechanism: The system should check for the presence of the lock file before starting a new registration operation on the same file. If present, the operation would be blocked or postponed.
